### PR TITLE
Allow multiple provenances for some hdk api functions.

### PR DESCRIPTION
**Pull request by struktured**
## PR summary

This PR features the ability to add [multiple provenances](https://miro.com/app/board/o9J_kyiXmFs=/?moveToWidget=3074457346592038454). It introduces a new generalized form of `hdk::commit_entry` named `commit_entry_result` to commit entries with additional provenance information, such as for countersigning purposes. It follows the same conventions as `get_entry_result`, providing extensible `CommitEntryOptions` and `CommitEntryResult` containers. The original `commit_entry` function calls the more advanced one but its interface signature was not changed.

## testing/benchmarking notes

- round trip unit test added in nucleus/ribosome/api/commit.rs
- `create_post_countersigned` app spec test added to blog zome

( if any manual testing or benchmarking was/should be done, add notes and/or screenshots here )

## followups

- extend `update_entry` and `remove_entry` to support additional options like `commit_entry_result` using the same style of extension.

## changelog

Please check one of the following, relating to the [CHANGELOG](https://github.com/holochain/holochain-rust/blob/develop/CHANGELOG.md)

- [x] this is a code change that effects some consumer (e.g. zome developers) of holochain core so there is an 'Unreleased' CHANGELOG item, with the format `- summary of change [PR#1234](https://github.com/holochain/holochain-rust/pull/1234)`
- [ ] this is not a code change, or doesn't effect anyone outside holochain core development

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Added
 - Adds hdk::grant_capability() for zome functions to be able to create [capability grant](doc/architecture/decisions/0017-capabilities.md)  [#1285](https://github.com/holochain/holochain-rust/pull/1285)
 
+- Adds hdk::commit_entry_result() which features: optional argument to include additional provenances. [#1320](https://github.com/holochain/holochain-rust/pull/1320)
```
Comment (zippy) [inline CHANGELOG.md]: Now need to move this to CHANGELOG_UNRELEASED.md as per @thedavidmeister request

File: core/src/nucleus/ribosome/api/commit.rs [hunk 2]
```diff
@@ -2,32 +2,40 @@ use crate::{
...
-use holochain_core_types::{cas::content::Address, entry::Entry, error::HolochainError};
+use holochain_core_types::error::HolochainError;
+
+use holochain_wasm_utils::api_serialization::commit_entry::{CommitEntryArgs, CommitEntryResult};
+
 use std::convert::TryFrom;
 use wasmi::{RuntimeArgs, RuntimeValue};
 
-/// ZomeApiFunction::CommitAppEntry function code
+/// ZomeApiFunction::CommitArgs function code
```
Comment (zippy) [inline core/src/nucleus/ribosome/api/commit.rs]: This seems like a mistaken change to the comment.

File: CHANGELOG.md (hunk 1 above)
Comment (struktured) [reply CHANGELOG.md]: Done in 2e56a3f771605079ba70fda7944cdbc5d3c80c8e

File: core/src/nucleus/ribosome/api/commit.rs (hunk 2 above)
Comment (struktured) [reply core/src/nucleus/ribosome/api/commit.rs]: Reverted back in 87733210074c12bca9177c4ad22e81a8cc35dfe7.

**Pull request merged**